devito/mpi/distributed.py

```diff
@@ -600,17 +616,53 @@ class CustomTopology(tuple):
     """
 
     def __new__(cls, items, input_comm):
+        # Keep track of nstars and already defined decompositions
         nstars = len([i for i in items if i == '*'])
-        if nstars > 0:
-            if input_comm.size % nstars != 0:
-                raise ValueError("Invalid `topology` for given nprocs")
-            if any(i not in ('*', 1) for i in items):
-                raise ValueError("Custom topology must be only 1 or *")
-
-            v = input_comm.size // nstars
-            processed = [i if i == 1 else v for i in items]
-        else:
+
+        # If no stars exist we are ready
+        if nstars == 0:
             processed = items
+        else:
+            # Init decomposition list
+            processed = [1] * len(items)
+
+            # Get star and integer indices
+            int_pos = [i for i, item in enumerate(items) if isinstance(item, int)]
```

Comment:
instead of int_pos and int_vals, and the loop below, why don't you just do it explicitely? it would be much neater

```
for i, item in enumerate(items):
    if isinstance(item, int):
         processed ...
```

etc etc

clearly move this after the `dd_list` (better name pls?) computation

Reply:
star_vals is better, idea from @mloubout